lib/galaxy/model/__init__.py

```diff
@@ -1422,6 +1422,22 @@ def dataset(self):
         return None
 
 
+# Set up output dataset association for export history jobs. Because job
+# uses a Dataset rather than an HDA or LDA, it's necessary to set up a
+# fake dataset association that provides the needed attributes for
+# preparing a job.
+class FakeDatasetAssociation (object):
```

Comment:
You can use this one in https://github.com/galaxyproject/galaxy/blob/972d641b1abec0117de812f0200eb3bad81f1822/lib/galaxy/tools/evaluation.py#L89 now to avoid duplication

Reply:
The one in galaxy/lib/galaxy/tools/evaluation.py should be already gone, isn't it?

Reply:
yup, I'm just blind ... :)